test/lib/jdk/test/lib/util/FileUtils.java

```diff
@@ -377,6 +378,35 @@ public static long getProcessHandleCount() {
         }
     }
 
+    /**
+     * Patches a part of a file.
+     * @param path of file
+     * @param fromLine the first line to patch. This is the number you see in an editor, 1-based.
+     * @param toLine the last line to patch. This is the number you see in an editor, inclusive.
+     * @param from the lines to remove, used to ensure the correct lines are removed. Ignored if null.
+     * @param to the newly added line, can be multiple lines or empty. Cannot be null.
+     * @throws IOException
+     */
+    public static void patch(Path path, int fromLine, int toLine, List<String> from, String to) throws IOException {
+        var lines = Files.readAllLines(path);
+        if (from != null) {
+            var removed = "";
+            for (int i = fromLine; i <= toLine; i++) {
+                removed += lines.remove(fromLine - 1).trim();
+            }
+            var froms = from.stream()
+                    .map(String::trim)
+                    .collect(Collectors.joining());
+            if (!removed.equals(froms)) {
+                throw new IOException("Removed not the same");
+            }
```

Comment:
That's a bit strange. I would suggest to return the removed lines instead, or to pass a `Consumer<String>` (or a Predicate<String> ?) that will accept the removed lines. You could continue to remove if the predicate returns true and throw if it returns false. It would also enable you tell exactly which line failed the check.

Reply:
I was just thinking about providing the removed lines and the added lines at the same time into the method. The exception here can probably be enhanced to show the content of `removed`. It's like writing 2 pieces of code if a consumer or predicate is needed. Here I've already trimmed the lines to make sure whitespaces do not matter.

Reply:
Shouldn't the comparison be better implemented by the caller then, who will know whether spaces are important or not? That's why I had suggested taking a Predicate<String> that could be called with each line removed, and the caller could interrupt the parsing by returning false when it detects a mismatch with what they expect.